lib/manager/gradle/build-gradle.ts

```diff
@@ -211,8 +223,14 @@ export function collectVersionVariables(
       }
     }
 
-    if (!dep.currentValue && variables[depName]) {
-      dep.currentValue = variables[depName];
+    if (!dep.currentValue) {
+      const dependencyLiteralRegex = dependencyStringLiteralExpressionFormatMatch(
+        dependency
+      );
+      const currentValue = dependencyLiteralRegex.exec(buildGradleContent)?.[1];
+      if (gradleVersioning.isValid(currentValue)) {
```

Comment:
We normally delay `isValid` checks until the process/lookup stage. Is it necessary to filter early here?

Reply:
Should not be a problem if we get rid of it here